### PR TITLE
Add kube-controllers deployment to Canal

### PR DESCRIPTION
**Pull request by lmm**
## Description

This PR modifies the Canal manifests so that kube-controllers is installed as well.
Kube-controllers is needed for auto host endpoints.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 30e52d43eb1238fbdb2ae7a0f0ca16651b425610

https://deploy-preview-3461--calico.netlify.app

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 0c0a76d2233e1566acffb542499636e2257d7f60

https://deploy-preview-3461--calico.netlify.app

File: _includes/charts/calico/templates/calico-kube-controllers.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if or (eq .Values.datastore "etcd") (eq .Values.network "calico") }}
+{{- if or (eq .Values.datastore "etcd") (eq .Values.network "calico") (eq .Values.network "flannel") }}
```
Comment (caseydavenport) [inline _includes/charts/calico/templates/calico-kube-controllers.yaml:1]: I think with this change we're saying that we run kube-controllers all the time, right? Do we need to limit to "flannel" || "calico" || "etcd"?

Also, what is the scenario where you're not using flannel, nor calico, but you are using etcd?

File: _includes/charts/calico/templates/calico-kube-controllers.yaml [hunk 2]
```diff
@@ -68,7 +68,11 @@ spec:
                   key: etcd_cert
             # Choose which controllers to run.
             - name: ENABLED_CONTROLLERS
+{{- if eq .Values.network "flannel" }}
```
Comment (caseydavenport) [inline _includes/charts/calico/templates/calico-kube-controllers.yaml:71]: I don't think this is quite right - we still want to run all these other controllers for flannel in etcd mode IIUC.

Comment (lmm) [reply _includes/charts/calico/templates/calico-kube-controllers.yaml]: Fixed

File: _includes/charts/calico/templates/calico-kube-controllers.yaml (hunk 1 above)
Comment (lmm) [reply _includes/charts/calico/templates/calico-kube-controllers.yaml]: Thanks, fixed

**Pull request merged**